stratum/client.cpp

```diff
@@ -126,15 +126,55 @@ bool client_subscribe(YAAMP_CLIENT *client, json_value *json_params)
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////
+bool client_user_address_already_validated(YAAMP_CLIENT *client)
+{
+	for(CLI li = g_list_client.first; li; li = li->next)
+	{
+		YAAMP_CLIENT *existingclient = (YAAMP_CLIENT *)li->data;
+		if (!strcmp(client->username, existingclient->username))
+		{
+			return true;
+		}		
+	}
+	return false;
+}
+
+/*
+ * Check if the clients username is already in the database
+ */
+bool client_validate_user_address_db(YAAMP_CLIENT *client)
+{
+	bool ret_val = false;
+	if (!g_db) return false;
+
+	db_query(g_db, "select exists (select * from accounts where username='%s');", client->username);
```

Comment:
i dont like this query... i prefer :

    SELECT id FROM accounts WHERE username='%s' AND is_locked=0

Reply:
I thought all addresses in the database were validated against a wallet at some time.
But I could be mistaken and more conditions are needed. 
My preference would be a boolean response (since you removed the exists).